### DISCUSSION TITLE
cobbler

### DISCUSSION THREAD

**Discussion by 1392273211** [cobbler/cobbler-web#219] [Q&A]
Can Cobbler provide ctyunos, which is based on Huawei Euler's iOS for Tianyi Cloud's iso? After installing Cobbler Cobbler web, the web page cannot function properly

**Comment by SchoolGuy** [comment]
Thanks for coming to the forums. Cobbler-Web is still in early development. As such a fully functioning web interface should not be expected for now.

Do you want to use Ctyunos as an operating system that Cobbler runs on or as an operating system that is being served for the purpose of PXE booting?

**Reply by 1392273211** [reply]
It means that after installing the cobbler cobbler web on ctyunos, but the web cannot be accessed, and various Baidu and Google chatGPTs cannot solve it. Moreover, ctyunos is equipped with the cobbler 3 package and the deployment 2 package. Therefore, I need to install cobbler on ctyunos and install the system in bulk

**Reply by 1392273211** [reply]
Boss, can the Cobbler Cobbler web community provide binary installation packages, or how to deploy Cobbler in the form of source code? How to compile and install the source code? There are no installation documents available online, but would it be convenient for you to write about them?

**Reply by SchoolGuy** [reply]
Cobbler until 3.2.2 had a Django based WebUI. This is bundled and the setup instructions are available via Read The Docs as all other documentation is.

The repository we are currently discussing in, is the new Angular based WebUI that is not ready for production yet. As such there are no instructions available so far.

**Reply by 1392273211** [reply]
Does the cobbler-web community provide binary installation packages or yum sources? It’s version 3.3

**Reply by SchoolGuy** [reply]
For Cobbler 3.3.x there is no functional WebUI at the moment.

**Reply by 1392273211** [reply]
3.3 does not have a UI interface, does it mean that 3.2 3.1 3.0 has a UI interface?

**Reply by opoplawski** [reply]
Yes, 3.2 and earlier have a web UI.

**Reply by InterceptorF** [reply]
Hey guys, 
I'm trying to install the developer version of your Angular cobbler-web. I am getting the following "unauthorized: " when trying to pull the docker image so I can look at making it work - I NEED that web GUI so I might as well try and work on it in my copious amounts of free time, and yes I am logged into redhat to pull this:
[ddn@cos-pxe-01 cobbler]$ docker run -it --rm --name cobbler-dev -p 80:80 -p 443:443 -v ${PWD}:/code cobbler-dev
Emulate Docker CLI using podman. Create /etc/containers/nodocker to quiet msg.
✔ registry.redhat.io/cobbler-dev:latest
Trying to pull registry.redhat.io/cobbler-dev:latest...
Error: initializing source docker://registry.redhat.io/cobbler-dev:latest: reading manifest latest in registry.redhat.io/cobbler-dev: unauthorized: access to the requested resource is not authorized

So I'm stuck. I have the web frame starting but it won't let me log in as cobbler/cobbler and I am assuming it's because the development version of Cobbler in a docker container  isn't starting.
I do have cobbler version 3.2.2 running on this server with two distros installed and would rather use that instead of a container:

[root@cos-pxe-01 ddn]# cobbler version
Cobbler 3.2.2
  source: ?, ?
  build time: Wed Apr  5 15:54:53 2023
[root@cos-pxe-01 ddn]# cobbler check
No configuration problems found.  All systems go.

[root@cos-pxe-01 ddn]# cobbler distro report
Name                           : rhel-8.8-x86_64
Architecture                   : x86_64
Automatic Installation Template Metadata : {'tree': 'http://@@http_server@@/cblr/links/rhel-8.8-x86_64'}
TFTP Boot Files                : {}
Boot loader                    : grub
Breed                          : redhat
Comment                        :
Fetchable Files                : {}
Initrd                         : /var/www/cobbler/distro_mirror/rhel-8.8-x86_64/images/pxeboot/initrd.img
Kernel                         : /var/www/cobbler/distro_mirror/rhel-8.8-x86_64/images/pxeboot/vmlinuz
Kernel Options                 : {}
Kernel Options (Post Install)  : {}
Management Classes             : []
OS Version                     : rhel8
Owners                         : ['admin']
Redhat Management Key          :
Remote Boot Initrd             : ~
Remote Boot Kernel             : ~
Template Files                 : {}

Name                           : gs-5.2.0.0-RHEL-r112-x86_64
Architecture                   : x86_64
Automatic Installation Template Metadata : {'tree': 'http://@@http_server@@/cblr/links/gs-5.2.0.0-RHEL-r112-x86_64'}
TFTP Boot Files                : {}
Boot loader                    : grub
Breed                          : redhat
Comment                        :
Fetchable Files                : {}
Initrd                         : /var/www/cobbler/distro_mirror/gs-5.2.0.0-RHEL-r112-x86_64/images/pxeboot/initrd.img
Kernel                         : /var/www/cobbler/distro_mirror/gs-5.2.0.0-RHEL-r112-x86_64/images/pxeboot/vmlinuz
Kernel Options                 : {}
Kernel Options (Post Install)  : {}


Thoughts?
Is there a way to port a running cobbler system to the new Angular cobbler-web without using the docker container?

**Reply by SchoolGuy** [reply]
@InterceptorF You are pulling from the wrong registry.

**Reply by SchoolGuy** [reply]
Either directly specify localhost if you want to use a locally built image or pull & specify the "user" one from ghcr.io.
